enhancements/storage/vsphere-storage-check.md

```diff
@@ -0,0 +1,228 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Below we propose to check if OpenShift clusters installed into vSphere environment is / can be fully functional by introducing a new operator and periodical checks.
+
+## Motivation
+
+Our support frequently faces OCP clusters on vSphere that is either mis-configured.
+
+* OCP lacks necessary permissions: [#1869035](https://bugzilla.redhat.com/show_bug.cgi?id=1869035) [#1879959](https://bugzilla.redhat.com/show_bug.cgi?id=1879959). [#1852304](https://bugzilla.redhat.com/show_bug.cgi?id=1852304).
+* Datastore used in storage classes / OCP config is too long: [#1884202](https://bugzilla.redhat.com/show_bug.cgi?id=1884202).
+* Users don't enable `disk.enableUUID=TRUE`: [#1434709](https://bugzilla.redhat.com/show_bug.cgi?id=1434709) [1850984](https://bugzilla.redhat.com/show_bug.cgi?id=1850984) [1727991](https://bugzilla.redhat.com/show_bug.cgi?id=1727991).
+* Kubelet is configured without `--cloud-provider=vsphere`: [1632729](https://bugzilla.redhat.com/show_bug.cgi?id=1632729).
+* Storage Distributed Resource Scheduler (DRS) is not supported.
+* vMotion (i.e. VM migration) is not supported.
+
+Such support cases are often frustrating, as there is high variety of vSphere environments, error messages we get from vSphere are not always useful and at the same time users either can't use the cluster (can't create / attach PVs) or even loose data.
+
+Secondary objective is that we (as OCP engineering / product management / support) do not know how our customers configure vSphere / OCP on top of it and if a future OCP release may break such OCP installation.
+
+* We'd like to know HW version of virtual machines they use (necessary for vSphere CSI driver & removing in-tree cloud provider).
+* We'd like to know what vSphere version they use (same as above).
+
+### Goals
+
+* User can detect and fix OCP / vSphere storage related configuration issues by themselves.
+* Our support can easily check OCP was correctly configured and supportable.
+* We get metrics about vSphere environments where OCP runs.
+* Add pre-flight checks to installer to fail installation if provided vSphere cluster / credentials are not suitable.
+
+### Non-Goals
+
+* Any automatic fix of the environment.
+
+## Proposal
+
+Run a dedicated vsphere-monitoring-operator that periodically tests OCP / vSphere:
+
+* The operator reports metrics, some of them sent via telemetry to us.
+* The operator reports alerts for common configuration issues.
+* The operator logs performed checks + more detailed info about OCP/vSphere config, so it's available in must-gather.
+
+Add pre-flight checks to installer.
+
+### Checks
+
+These checks will be performed regularly and / or in the installer:
+
+| Test | Periodic | Installer |
+|-|-|-|
+| OCP can list recent tasks. | yes | yes |
+| OCP can list folders in the default datastore. |  yes | yes |
```

Comment:
These both seem like reasonable things to check in the installer in the "Platform Permissions Check" asset.